src/features/services/components/ledger-settings.tsx

```diff
@@ -38,9 +38,24 @@ function TokenRow({ name, symbol, address }: Token) {
   );
 }
 
+function toToken(token: TokenInfo): Token {
+  return {
+    name: token.info.summary.name,
+    address: token.info.address.toString(),
+    symbol: token.info.summary.symbol,
+  };
+}
+
 export function LedgerSettings() {
   const account = useAccountsStore((s) => s.byId.get(s.activeId));
-  const { data, isError, isLoading } = useTokenInfo();
+  const tokenList = useTokenList();
+  const tokenInfo = useTokenInfo(tokenList);
+
+  const isLoading = tokenInfo.some((q) => q.isLoading);
+  const isError = tokenInfo.some((q) => q.isError);
```

Comment:
Since `tokenInfo` is an array of queries, we do a little magic here to make the error / loading components work.